core/lib/resources/presence/presence_manager.js

```diff
@@ -118,8 +126,14 @@ PresenceManager.prototype.stampExpiration = function(message) {
   message.at = Date.now() + PresenceManager.messageExpiry;
 };
 
+// FIXME: Method signature is getting unmanageable.  
 PresenceManager.prototype.addClient = function(socketId, userId, userType,
-                                                      userData, callback) {
+                                                      userData, clientData, callback) {
+
+  if (typeof(clientData) === 'function') {
+    callback = clientData;
+  }
```

Comment:
Some tests are using are calling PresenceManager.addClient directly. This guard addresses that issue. Should we change the tests instead? We don't know how people might be using this.